dts/bindings/pinctrl/nuvoton,numaker-pinctrl.yaml

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Nuvoton Technology Corp. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Pin controller is responsible for controlling pin function
+  selection and pin properties. For example, for example you can
+  use this node to set UART0 RX as pin PB12 to fulfill SYS_GPB_MFP3_PB12MFP_UART0_RXD.
+
+  The node has the 'pinctrl' node label set in your SoC's devicetree,
+  so you can modify it like this:
+
+    &pinctrl {
+            /* your modifications go here */
+    };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+    &pinctrl {
+      /* configuration for the uart0 "default" state */
+      uart0_default: uart0_default {
+          /* configure PB13 as UART0 TX and PB12 as UART0 RX */
+          pinmux = <PB12MFP_UART0_RXD 0x0000>, <PB13MFP_UART0_TXD 0x0000>;
+      };
+    };
+
+
+  To link pin configurations with a device, use a pinctrl-N property for some
+  number N, like this example you could place in your board's DTS file:
+
+    #include "board-pinctrl.dtsi"
+
+    &uart0 {
+        pinctrl-0 = <&uart0_default>;
+        pinctrl-names = "default";
+    };
+
+compatible: "nuvoton,numaker-pinctrl"
+
+include:
+  - name: base.yaml
+  - name: pincfg-node.yaml
+    child-binding:
+      property-allowlist:
+        - drive-push-pull
+        - drive-open-drain
+        - bias-disable
+        - bias-pull-down
+        - bias-pull-up
+
+child-binding:
+  description: |
+    Each child node defines the configuration for a particular state.
+  properties:
+    pinmux:
+      required: true
+      type: array
```

Comment:
why is this an array? I see second field is always `0`

Reply:
For example uart/serial port, it may need TX/RX/CTS/RTS, so it's an array.
like as:
```
	uart1_default: uart1_default {
		pinmux = <PB9MFP_UART1_nCTS 0x0000>,
			 <PB8MFP_UART1_nRTS 0x0000>,
			 <PB2MFP_UART1_RXD 0x0000>,
			 <PB3MFP_UART1_TXD 0x0000>,
```

Reply:
sorry, I meant the `0x0000`

Reply:
it's the reserved config data for some complex NuMaker chip.
For M46x, 1st field 'PB9MFP_UART1_nCTS' is enough to encode all pinmux information.
For some complex NuMaker chip, one field can't encode all information.
Sorry, it's my consideration for compatible all NuMaker chips in near future.

Reply:
Note: fields are 32-bit maximum, so you can just let e.g. `PB9MFP_UART1_nCTS` use 32-bit, with unused 16-bit. This way you don't need this 16-bit+16-bit(all-zero) pattern, it looks weird. Can you please justify a bit this "future requirements"? Why a single 32-bit field is not an option?

Reply:
Sorry, 0x0000 is a little confused, in fact, both of 1st field (i.e PB9MFP_UART1_nCTS) and 2nd field are 32-bit.  It will be better as <PB9MFP_UART1_nCTS 0x00000000>.
Current Bit[31-28] for pin group, Bit[27-24] for pins of one group. As while groups over 16 or pins of one group over 16, it will need 2nd field.

Reply:
After discussion with H/W architecture designer, it's feasible to limit MCU multi-function pins under 16*16 criteria in the future design. So, I will remove 2nd field 0x0000.

Reply:
Thanks of your review.
Finished the modification of pin-mux with a single 32-bit field.

Reply:
Hi @gmarull ,
Done and force pushed branch for this modification of pin-mux.

Reply:
Hi @gmarull ,
  Please have a look into this update.